buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/unittestandcompile/UnitTestAndCompilePlugin.kt

```diff
@@ -221,7 +235,7 @@ class UnitTestAndCompilePlugin : Plugin<Project> {
     fun Test.onlyRunPreviousFailedClassesIfNecessary() {
         if (project.stringPropertyOrEmpty("onlyPreviousFailedTestClasses").toBoolean()) {
             val previousFailedClasses = getPreviousFailedTestClasses()
-            if (previousFailedClasses.isEmpty()) {
+            if (previousFailedClasses.isEmpty() || allTestFailuresCount.get() > tooManyTestFailuresThreshold) {
```

Comment:
This might be not accurate enough. For example, a build contains two test tasks, test task A has 8 failures, test task B has 6 failures, then both A and B should be disabled, but this logic will only disable test task B. I guess this is fine, we can adjust this logic and the threshold 10 if there's evidence they're not good enough.